UserAdminBundle/Controller/Api/UserController.php

```diff
@@ -26,20 +26,22 @@
 class UserController extends BaseController
 {
     /**
-     * @param string $userId
+     * @param string $email
      *
-     * @Config\Route("/{userId}", name="open_orchestra_api_user_show")
+     * @Config\Route("/{email}", name="open_orchestra_api_user_show")
      * @Config\Method({"GET"})
      *
      * @return FacadeInterface
      */
-    public function showAction($userId)
+    public function showAction($email)
     {
-        $user = $this->get('open_orchestra_user.repository.user')->find($userId);
+        $this->denyAccessUnlessGranted(ContributionActionInterface::READ, UserInterface::ENTITY_TYPE);
+        $user = $this->get('open_orchestra_user.repository.user')->findOneByEmail($email);
```

Comment:
@alavieille : can you check there is a user before passing it to the transformer ? As the email comes from the url there's no warranty that the email match a user.